server/expression/in_tuple.go

```diff
@@ -85,6 +85,9 @@ func (it *InTuple) Eval(ctx *sql.Context, row sql.Row) (any, error) {
 		if err != nil {
 			return nil, err
 		}
+		if result == nil {
+			return nil, nil
+		}
```

Comment:
This is incorrect, and would cause the following query to return the wrong result:
```sql
SELECT 4 IN (1, 2, 3, null::int4, 4); -- should be true, but the above changes return null
```
You are onto something here though.
https://www.postgresql.org/docs/15/functions-subquery.html
> Note that if the left-hand expression yields null, or if there are no equal right-hand values and at least one right-hand row yields null, the result of the IN construct will be null, not false. This is in accordance with SQL's normal rules for Boolean combinations of null values.

So we should replicate the below behavior:
```sql
SELECT 4 IN (1, 2, 3, null::int4); -- null
SELECT 4 IN (1, 2, 3); -- false
```

Reply:
Fixed